module/zfs/arc.c

```diff
@@ -928,7 +928,7 @@ static unsigned long l2arc_rebuild_blocks_min_l2size = 1024 * 1024 * 1024;
 
 /* L2ARC persistence rebuild control routines. */
 void l2arc_rebuild_vdev(vdev_t *vd, boolean_t reopen);
-static _Noreturn void l2arc_dev_rebuild_thread(void *arg);
+static __attribute__((noreturn)) void l2arc_dev_rebuild_thread(void *arg);
```

Comment:
Also, this maybe no harm to foul, but I do wonder that with forward `static` declarations that use `__attribute__((noreturn))` is it necessary to also put this in the function definition signature as well? Honestly, I don't think it matters, but I did notice that this was done in some areas but not in others. Just wondering if we would want to make this consistent across everything either with it included in both the declaration and definition signature or only in the declaration?

Reply:
I'm not sure it's needed, but I opted to try and keep both declarations consistent in this case.  I think we'd want to make everything consistent in both places, but I didn't tackle any of this in this PR.